pegasus/sites.v3/hourofcode.com/views/signup_form.haml

```diff
@@ -87,14 +106,55 @@
         $('#hoc-special-event-details').closest('.form-group').slideUp();
       }
     });
+
     $('#hoc-event-type').change(function() {
-      if($(this).val() == 'in_school') {
-        $('#hoc-entire-school-flag').closest('.form-group').show();
+      if(($(this).val() == 'in_school') && ($("#country").val() == 'US')) {
+        $('#usa-school-info').show();
+        $('#non-usa-event-info').hide();
+        $('#census-questions').show();
       } else {
-        $('#hoc-entire-school-flag').closest('.form-group').hide();
+        $('#usa-school-info').hide();
+        $('#non-usa-event-info').show();
+        $('#census-questions').hide();
       }
     });
-  });
+
+    $('#country').change(function() {
```

Comment:
This function is very similar to the one above, but I need to ensure that when either field changes we check to see what should be shown/hidden. Is there are cleaner way to do this?

Reply:
Can you factor out the common code into a standalone function and parameterise it for its differences?